docs/modules/ROOT/pages/adr/0042-backup-encryption-for-mariadb-galera-and-postgresql.adoc

```diff
@@ -0,0 +1,167 @@
+
+The s3proxy router implementation uses the provided host header as the endpoint, requiring DNS hijacking to redirect requests to our S3-compatible storage backend. See the official example for details.
+
+[NOTE]
+s3proxy is licensed under the Business Source License 1.1, which prohibits production use without a commercial license
+
+==== References
+
+- https://docs.edgeless.systems/constellation/getting-started/examples/filestash-s3proxy[Offical example]
+- https://github.com/edgelesssys/constellation/blob/main/s3proxy/internal/router/router.go#L267[s3proxy router implementation]
+- https://github.com/edgelesssys/constellation/blob/main/s3proxy/deploy/s3proxy/values.yaml[Helmchart values]
+
+=== Rclone - recommended
+
+Rclone is a command-line program for managing files on cloud storage, supporting various backends including S3-compatible storage. Using `rclone serve` with crypt remotes, we can implement a transparent encryption proxy.
+
+The configuration requires two remotes: the first is the actual storage backend, and the second is a crypt remote that wraps the first. The crypt remote handles encryption and decryption transparently.
+
+[source,toml]
+----
+[minio]
+type = s3
+provider = Minio
+env_auth = false
+access_key_id = minioadmin
+secret_access_key = minioadmin
+endpoint = http://127.0.0.1:9000
+region = us-east-1
+location_constraint = us-east-1
+force_path_style = true
+
+[s3-crypt]
+type = crypt
+remote = minio:backups <1>
+filename_encryption = standard
+directory_name_encryption = true
+password = IRv0dJ285I6uxOAKW7tpwirL9TxiF5wbwAlSHA <2>
+password2 = rPRcLTS8YCbzeV_WiDfrB-fhf5f1dGJ94CXBzw <3>
+----
+<1> Use the minio remote as the storage backend
+<2> Password used for encryption
+<3> Password used for salt
+
+We can then run rclone with: `rclone serve s3 s3-crypt: --addr :9090 --auth-key proxyaccess,proxysecret`.
+
+Rclone is the most viable option: it's actively maintained, widely used, and has straightforward configuration.
+
+==== References
+
+- https://rclone.org/crypt/[Rclone crypt]
+- https://rclone.org/commands/rclone_serve_s3/[Rclone serve s3]
+- https://rclone.org/commands/rclone_serve_s3/[Rclone s3 backend]
+
+
+[NOTE]
+====
+**Key rotation**
+
+Since rclone uses the provided password to generate the encryption key, changing the password is not possible for already encrypted data. The data would need to be wiped from the remote and re-encrypted with the new key. This is not feasible in our setup. Since our setup with k8up also doesn't do key rotation, this isn't an issue.
+====
+
+==== Storing encryption passwords
+
+Since we don't plan to rotate the password used for the encryption key, the password can be saved inside of a Kubernetes secret. This is also how we do it with the repository password for K8up.
```

Comment:
Just to keep in mind, password rotation could be required should we somehow leak the password or be hacked.